lib/javascript/JavascriptParser.js

```diff
@@ -4271,23 +4271,23 @@ class JavascriptParser extends Parser {
 
 	/**
 	 * @param {MemberExpression} expression a member expression
-	 * @returns {{ members: string[], object: Expression | Super, membersOptionals: boolean[], memberRangeStarts: number[] }} member names (reverse order) and remaining object
+	 * @returns {{ members: string[], object: Expression | Super, membersOptionals: boolean[], memberRanges: Range[] }} member names (reverse order) and remaining object
 	 */
 	extractMemberExpressionChain(expression) {
 		/** @type {AnyNode} */
 		let expr = expression;
 		const members = [];
 		const membersOptionals = [];
-		const memberRangeStarts = [];
+		const memberRanges = [];
 		while (expr.type === "MemberExpression") {
 			if (expr.computed) {
 				if (expr.property.type !== "Literal") break;
-				members.push(`${expr.property.value}`);
-				memberRangeStarts.push(expr.object.range[1]);
+				members.push(`${expr.property.value}`); // the literal
+				memberRanges.push(expr.object.range); // the range of the expression fragment before the literal
```

Comment:
This is Important Part 1 of the change.  Capture the entire subrange and not just the end.